packages/cursorless-engine/src/languages/TreeSitterQuery/TreeSitterQuery.ts

```diff
@@ -83,7 +84,25 @@ export class TreeSitterQuery {
         this.patternPredicates[match.patternIdx].every((predicate) =>
           predicate(match),
         ),
-      );
+      )
+      .map((match): QueryMatch => {
+        // Merge the ranges of all captures with the same name into a single
+        // range and return one capture with that name.  We consider captures
+        // with names `@foo`, `@foo.start`, and `@foo.end` to have the same
+        // name, for which we'd return a capture with name `foo`.
+        const captures: QueryCapture[] = Object.entries(
+          groupBy(match.captures, ({ name }) =>
+            name.replace(/\.(start|end)$/, ""),
+          ),
+        ).map(([name, captures]) => ({
+          name,
+          range: captures
+            .map(({ range }) => range)
+            .reduce((accumulator, range) => range.union(accumulator)),
```

Comment:
Is this too lenient?  I'm not sure we want to allow people to use multiple `@foo` captures and rely on the fact they're merged.  As discussed in Slack, I don't think the semantics are obvious when you see multiple `@foo` captures; in fact I'd expect multiple targets.  I could add a check that ensures there's only one `@foo` etc

Reply:
I'm fine with a check. The waits implemented now you could mix and match start and end and it wouldn't matter.